### DISCUSSION TITLE
Two-dimensional virtualization for displaying large data sets

### DISCUSSION THREAD

**Discussion by DarthMazut** [AvaloniaUI/Avalonia#12114] [General]
Hello,

Recently, I needed a 2D virtualization solution in Avalonia but couldn't find anything with satisfactory performance. Nesting one-dimensional panels with virtualization didn't provide the desired results, and even the solution I tried from [this repository](https://github.com/wieslawsoltes/Spreadsheet) felt a bit laggy. As a result, I developed my own approach to tackle this problem.

I created a repository called [VirtualizedGridAvalonia](https://github.com/DarthMazut/VirtualizedGridAvalonia), which now hosts the VirtualizedGrid control. I've documented the essential details in the project readme file, so please read it first if you're interested in this project: [Project Readme](https://github.com/DarthMazut/VirtualizedGridAvalonia#readme)

For your convenience, I've attached a video demonstration.

Your feedback is crucial to me. I'd like to know if this project has any value for you and whether I should continue developing it to make it more customizable beyond my own needs. I appreciate any feedback you can provide. Also, please don't hesitate to let me know if you believe the solution is fundamentally flawed. Your input will help me save time.

Thank you.

https://github.com/AvaloniaUI/Avalonia/assets/29342390/c56c9d3a-74d8-4549-88d6-72e1dd9e6250

**Comment by timunie** [comment]
Nice one. Consider to add it to AwsomeAvalonia
